Blended Measurement Paper

TallJimbo opened the pull request:
I've put together a first draft of a small paper on blended measurement as suggested by @mjuric at https://community.lsst.org/t/baseline-plan-for-blended-fitting/218/14

It's not really written at a consistent level - it assumes some LSST-specific jargon, but then doesn't explain some algorithmic background I didn't want to get into here (it's hard to draw the line).  But I think think it should be readable enough for most interested parties to comment on the algorithmic plan and whether the detail level is appropriate for DM design documents.

There are no references yet (it'd mostly have to reference the infamous Photo non-paper anyway).  There are clearly some I need to add, but I figure that's a lower priority than writing up other parts of the DRP design.

TallJimbo committed:
Start paper on deblended deep measurement.

TallJimbo committed:
More work on blended measurement paper.

TallJimbo committed:
Complete first draft of blended measurement paper.

Now includes figure using Confluence Gliffy with SVG output
translated to PDF using Inkscape (manually).  I've committed
only the PDF figure and the Gliffy native file.

ktlim commented on blended-measurement/blended-measurement.tex:
```diff
@@ -0,0 +1,685 @@
+
+\section{Introduction}
+
+Most LSST objects will overlap one or more of its neighbors enough to affect
```
Would be good to cite the requirements (i.e. DPDD) for what this measurement is intending to produce.

TallJimbo replied:
Done on 3bde83eb4e9891f00bfdbb27b130956e741bf90f.

ktlim commented on blended-measurement/blended-measurement.tex:
```diff
@@ -0,0 +1,685 @@
+described in Section~\ref{sec:deblender}, likely using a symmetry ansatz to
+define templates.  The inputs will be a detection catalog containing merged
+{\texttt Footprints} and {\texttt Peaks} from all detection images [D1], and
+at least one coadd image per filter [D2].  We may have multiple input coadds
```
Thinking worst cases: Is it possible that we won't have a coadd image or coadd detections corresponding to single-frame detections?  What if we're early in Commissioning or the seeing in the few frames we have wasn't good enough to meet the coadd threshold or something?

TallJimbo replied:
I think these cases should be rare enough that it will be acceptable to just ignore regions of the sky for which we don't have _any_ coadd-quality data.  I also think it's quite likely that we'll define "coadd-quality" in a relative and/or local sense, so if we have one visit in a region it'll always be considered better than nothing.

A trickier problem is when we only have coadd-quality data in some filters, but not all.  We'll need to make sure our algorithms still converge when some filters are missing or extremely low quality, and our masking and selection function characterization will have to indicate this somehow.

ktlim commented on blended-measurement/blended-measurement.tex:
```diff
@@ -0,0 +1,685 @@
+from different filters. The details of these inputs and the parallelization
+and data flow within the deblender itself are beyond the scope of this
+document. The outputs are deblended pixel values for both direct [D4] and PSF-
+matched [D5] coadds (generated by sequentially replacing neighbors with noise,
```
I thought the `HeavyFootprint` deblended pixel values do _not_ require noise replacement; it's the measurements in the next paragraph that need that.

TallJimbo replied:
I was trying to simplify the flowchart here by rolling noise replacement into the deblender, so the "deblended pixel values" I'm referring to here are actually the noise-replaced images (since producing the noise-replaced images is all we use the `HeavyFootprints` for).  If that was confusing, I should probably just make the noise replacement nodes explicit.

ktlim commented on blended-measurement/blended-measurement.tex:
```diff
@@ -0,0 +1,685 @@
+coadds and single-epoch images, requiring completely new algorithms for all
+operations performed on them.  As a result, making use of likelihood coadds
+may require considerably more human effort, but it could reduce the need for
+multi-epoch processing (but probably not eliminate it). If they prove viable,
```
How much investigation do we need to do to determine if they are viable, necessary, or better?  Do we have to build everything and then have a bake-off, or can we find out in a faster way?

TallJimbo replied:
I think it's very likely we'll at least use likelihood coadds for detection, though these could be made more cheaply (both in computational effort and implementation effort) than what we'd want to use for deblending and measurement.  So we'll want to implement at least that much.

Beyond that, there are three stages of tests I think we need to go through to see whether it's worth going further:
- Do some pencil-and-paper math and toy implementation coding to come up with ideas on how we'd even go about doing deblending and measurement on likelihood coadds.
- Run some experiments on simple simulations with a toy implementation to verify quality and get a rough idea of computational costs.
- Full implementation and testing.

We'd be able to give up on the idea (or some aspects) of it at any of these stages.

Should I work that into the writeup, or is that sort of discussion best left to DLP issue text?

ktlim commented on blended-measurement/blended-measurement.tex:
```diff
@@ -0,0 +1,685 @@
+PSF-matched coadds instead of direct coadds, even though direct coadds may
+allow the model parameters to be constrained better.
+
+If fitting to PSF-matched coadds produces better galaxy colors, we will do
```
Again, how can we know without building both and comparing?  Would we (emphasizing "we") do this, even if it were better, if we were already meeting SRD specs without it?

TallJimbo replied:
I think here we have to implement both and compare.  There are no SRD specs in this area to my knowledge, and in fact the only real measures of quality are things whose implementation is outside project scope (e.g. photometric redshift accuracy).  We almost certainly want to collect some code from science collaboration members to be able to do those tests ourselves, of course.

jdswinbank commented on blended-measurement/blended-measurement.tex:
```diff
@@ -0,0 +1,685 @@
+and parallax.  There are two distinguishing factors between these and faster
+solar system objects from an algorithmic perspective:
+\begin{itemize}
+\item They will be blended with the same neighbors in every epoch.  As a
```
Is this always true? It doesn't seem obvious.

TallJimbo committed:
Update flowchart pdf from svg.

jdswinbank committed:
Typo fixes.

TallJimbo commented:
@mjuric @rhl We (@jdswinbank and I) are starting to work on turning the approach described in this document into a plan in JIRA.  If you've got any concerns with what I've presented here as a new baseline (note that it *does* differ from the current DPDD, not just expand on it), we'd like your comments as soon as possible.

mjuric commented:
@TallJimbo et al.

I think the described approach is an improvement in terms of detail over what we have right now; I'm OK with you and @jdswinbank beginning to use it for planning purposes (it is less likely to underestimate the true work needed less than the current baseline).

We'll iterate on the details -- I still haven't had the time to fully digest it -- and to patch the DPDD and LDM-151 accordingly. Also, since virtually everything of importance will be blended in LSST to some extent, this paper is effectively a roadmap for all measurement (so I'll need to socialize it with our SciCollab colleagues once we've exhausted all questions).

The thing that worries me the most right now is using multi-band information to derive deblending templates, and simultaneous fitting in multiple bands. Those will make it impossible to derive (for example) independent shear estimates in (for example) g and r bands.

The thing that should worry @ktlim and @timj are expanded hardware requirements (i.e., I'm not sure if we've modeled the storage for parents, of it we've modeled storing covariance matrices for joint best fits).

PS: It would be good to understand what are (the various groups in) CFHT-LS and (the various groups in) DES doing. Is it similar to what you're proposing?

ktlim commented:
I'm rather less worried about additional storage requirements (although Heavy Footprints are somewhat of a concern).  I'm more worried about additional compute requirements for DRP (some of DavidN's suggestions at https://community.lsst.org/t/baseline-plan-for-blended-fitting/218/17?u=ktl seem like they might add factors of a few to the multifit computation requirements) and to a lesser extent compute and bandwidth requirements in the Data Access Centers.

TallJimbo commented:
> The thing that worries me the most right now is using multi-band information to derive deblending templates, and simultaneous fitting in multiple bands. Those will make it impossible to derive (for example) independent shear estimates in (for example) g and r bands.

The use case for independent shear estimates in different bands is just systematics cross-checks, I assume?

At least in the sampling stages, it should be possible to actually store the per-filter likelihoods, as well as their sum, which would make it possible to use importance sampling to split produce per-filter measurements in catalog space.  That should be possible for optimizer-based fitting at first order, though I doubt first order is good enough here.  I do think we will get the best photometry by considering all bands together, but of course that statement needs to be verified.

mjuric commented:
> The use case for independent shear estimates in different bands is just systematics cross-checks, I assume?

Correct. And the concerns that band-to-band morphology changes that are not captured by the model could leak into the shear signal (e.g., if blue galaxies are distributed differently than the red ones, with systematics injected preferentially to the blue sample due to their more complicated morphology, that may generate spurious shear signal), or just add noise.

My initial DPDD baseline proposal had what you're proposing (and for the same reasons -- better signal), but after gathering comments from the co-authors I switched it to independent per-band fits. I believe Tony T. was the biggest proponent of doing it so. If the DESC thinking has evolved, we can consider changing it again (IIRC, doing all bands at one is computationally cheaper). But this is a relatively big change so it would be good to understand what is the broad consensus on the best way of doing it. What are you doing for HSC?

TallJimbo commented:
On HSC, we're not doing anything sophisticated enough for this to be a meaningful questions (and we may not until LSST-side code is available to do it).  But I'm pretty sure one of the two DES shear pipelines (Erin Sheldon's ngmix) does fit multiple band simultaneously, and that's been described as an advantage.

I'm pretty sure there'd be a broad consensus among DESC members on wanting to be able to run shear measurement separately in different bands, but I also think there'd be a consensus on wanting to combine across bands once we'd verified that the results were consistent.  There may be differences as to whether the verification needs to happen on the full dataset or a subset, and whether the combination across bands needs to happen in the course of pixel-level modeling (I'm not sure it does for shear myself - I mostly think that's useful for photometry).

I don't *think* the danger of having biases from insufficiently flexible models vs. realistic morphology are qualitatively different in multi-band fitting, but I could be ignoring something important.   In both multi-band and single-band fitting, there's going to be some level of model bias, and we either need to get it small enough that we can ignore it or calibrate it away via sims or deep data.  I don't think redshift-dependent biases are any more of a concern here, either - that's already something we'll have to worry about in single-band fitting just because morphologies are evolving and SNR is going to be a strong function of redshift.

TallJimbo committed:
Merge pull request #4 from lsst/blended-measurement-jds

Minor cleanups and typo fixes.

TallJimbo committed:
Tie measurement algorithm mentions to DPDD.

TallJimbo committed:
Add some statements indicating how we can evaluate various options.

These evaluation plans need quite a bit more development, but we don't have
time for that now.

TallJimbo committed:
Add readme file explaining existence of blended measurement doc.

TallJimbo committed:
Include the compiled PDF for blended measurement in the repo.
